### PR TITLE
Feat create params middleware

### LINKED ISSUE #10: Create a middleware to parse request params into a map[string]string
Like we did with the issue here #8 we need a way to be able to parse request params as well and make it available in the request context. 
We can always do `mux.Vars(r)` as the time but surely that's a lot of repetition...and a middleware with help keep things structured. 

Also, we'll need one for `ParamsF` and `ParamsH`

### PR DESCRIPTION
**Pull request by Funmi4194**

### CODE REVIEW

File: middleware/v1/params.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/opensaucerer/gotemplate/typing"
+)
+
+// ParamsF is a middleware that takes a request handler.
```
Comment (opensaucerer) [inline middleware/v1/params.go]:
```go
// ParamsH is a middleware that takes a request handler
// and pushes the request path parameters it into the request's context
// using the ParamsCtxKey.
```

File: middleware/v1/params.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+			ctx := context.WithValue(r.Context(), typing.ParamsCtxKey{}, params)
+
+			//call next handler
+			next.ServeHTTP(w, r.WithContext(ctx))
+
+		})
+	}
+}
+
+// ParamsF is a middleware that takes a controller function.
```
Comment (opensaucerer) [inline middleware/v1/params.go]:
```go
// ParamsF is a middleware that takes a controller function
// and pushes the request path parameters into the request's context
// using the ParamsCtxKey.
```

File: middleware/v1/params.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+)
+
+// ParamsF is a middleware that takes a request handler.
+// It sets a variable paramas and pushes it into the request's context
+// using the ParamsCtxKey.
+func ParamsH() func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			//set variable params
+			params := mux.Vars(r)
```
Comment (opensaucerer) [inline middleware/v1/params.go]: We can prevent unnecessary garbage collection here by removing the `params` variable
```go
// set params to the request context
ctx := context.WithValue(r.Context(), typing.ParamsCtxKey{}, mux.Vars(r))
```

File: middleware/v1/params.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+// ParamsF is a middleware that takes a controller function.
+// It sets a variable paramas and pushes it into the request's context
+// using the ParamsCtxKey.
+func ParamsF() func(func(http.ResponseWriter, *http.Request)) http.Handler {
+	return func(next func(http.ResponseWriter, *http.Request)) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			//set variable params
+			params := mux.Vars(r)
```
Comment (opensaucerer) [inline middleware/v1/params.go]: We can prevent unnecessary garbage collection here by removing the `params` variable
```go
// set params to the request context
ctx := context.WithValue(r.Context(), typing.ParamsCtxKey{}, mux.Vars(r))
```

**Review by opensaucerer — CHANGES_REQUESTED** [review]
Comments

File: middleware/v1/params.go (hunk 1 above)
Comment (Funmi4194) [reply middleware/v1/params.go]: Okayy

File: middleware/v1/params.go (hunk 3 above)
Comment (Funmi4194) [reply middleware/v1/params.go]: Okayy.

**Review by opensaucerer — APPROVED** [review]
LGTM

**Pull request merged**